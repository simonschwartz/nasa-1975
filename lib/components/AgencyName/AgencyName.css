.NASA1975-agency-name {
  line-height: 1;
}

.NASA1975-agency-name-N,
.NASA1975-agency-name-Aa,
.NASA1975-agency-name-SA,
.NASA1975-agency-name-S,
.NASA1975-agency-name-A {
  display: block;
}

.NASA1975-agency-name--inline .NASA1975-agency-name-Aa {
  white-space: nowrap;
}

@media (min-width: 550px) {
  .NASA1975-agency-name--inline .NASA1975-agency-name-N,
  .NASA1975-agency-name--inline .NASA1975-agency-name-Aa,
  .NASA1975-agency-name--inline .NASA1975-agency-name-S,
  .NASA1975-agency-name--inline .NASA1975-agency-name-A {
    display: inline-block;
  }
}
