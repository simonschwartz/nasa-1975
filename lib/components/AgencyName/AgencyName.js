import React from "react";
import Heading from "../Heading";
import PropTypes from "prop-types";
import "./AgencyName.css";

const AgencyName = ({
  stacked = false,
  headingLevel = "1",
  size = "lg",
  centerName = "",
  className = "",
  ...attributeOptions
}) => {
  const headingClass = stacked
    ? "NASA1975-agency-name NASA1975-agency-name--stacked"
    : "NASA1975-agency-name NASA1975-agency-name--inline";

  return (
    <div className={className} {...attributeOptions}>
      <Heading
        level={headingLevel}
        size={size}
        weight="light"
        className={headingClass}
      >
        <span className="NASA1975-agency-name-N">National</span>{" "}
        <span className="NASA1975-agency-name-Aa">Aeronautics and</span>{" "}
        <span className="NASA1975-agency-name-SA">
          <span className="NASA1975-agency-name-S">Space</span>{" "}
          <span className="NASA1975-agency-name-A">Administration</span>
        </span>
      </Heading>
      {centerName &&
        <Heading level={headingLevel} size={size} weight="medium">
          {centerName}
        </Heading>}
    </div>
  );
};

AgencyName.propTypes = {
  /** Size of the heading */
  stacked: PropTypes.bool,
  /** HTML heading level */
  headingLevel: PropTypes.oneOf(["1", "2", "3", "4", "5"]),
  /** Size of the heading */
  size: PropTypes.oneOf(["sm", "md", "lg"]),
  /** Center name eg "John F. Kennedy Space Center" */
  centerName: PropTypes.string,
  /** className to add to heading */
  className: PropTypes.string
};

export default AgencyName;
