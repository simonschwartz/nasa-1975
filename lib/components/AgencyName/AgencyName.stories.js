import React from "react";
import AgencyName from "./";

/**
 * Docs page configuration
 */
export default {
  title: 'Agency Name',
  component: AgencyName,
  parameters: {
    componentSubtitle: 'Agency and Center Identification',
  },
};

/**
 * Component stories
 */
export const overview = () => (
  <AgencyName centerName="John F. Kennedy Space Center" />
);

export const stacked = () => (
  <AgencyName stacked />
);

export const sizesWithOutCenterName = () => (
  <div>
    <AgencyName size="xl" style={{ marginBottom: "2rem" }} />
    <AgencyName size="lg" style={{ marginBottom: "2rem" }} />
    <AgencyName size="md" style={{ marginBottom: "2rem" }} />
    <AgencyName size="sm" style={{ marginBottom: "2rem" }} />
    <AgencyName size="xs" />
  </div>
);

export const sizesWithCenterName = () => (
  <div>
    <AgencyName size="xl" centerName="John F. Kennedy Space Center" style={{ marginBottom: "2rem" }} />
    <AgencyName size="lg" centerName="John F. Kennedy Space Center" style={{ marginBottom: "2rem" }} />
    <AgencyName size="md" centerName="John F. Kennedy Space Center" style={{ marginBottom: "2rem" }} />
    <AgencyName size="sm" centerName="John F. Kennedy Space Center" style={{ marginBottom: "2rem" }} />
    <AgencyName size="xs" centerName="John F. Kennedy Space Center" style={{ marginBottom: "2rem" }} />
  </div>
);
