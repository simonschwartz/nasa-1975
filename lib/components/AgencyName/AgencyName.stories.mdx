import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import AgencyName from './';
import StoryContainer from '../../../.storybook/components/StoryContainer';

<Meta title="Agency Name" component={AgencyName} />

# Agency Name

The full agency name (National Aeronautics and Space Administration) is used, usually in conjunction
with the worm logo, to identify the agency as a total entity.

<div style={{ margin: '3rem 0' }}>
    <AgencyName size="xl" style={{ marginBottom: '2rem' }} />
    <AgencyName size="lg" style={{ marginBottom: '2rem' }} />
    <AgencyName size="md" style={{ marginBottom: '2rem' }} />
</div>

### Stacked

The agency name can be displayed stacked.

<StoryContainer>
    <AgencyName size="md" stacked />
</StoryContainer>

### Inline

The agency name is displayed inline by default.

<StoryContainer>
    <AgencyName size="md" />
</StoryContainer>

### Center Identification

To identify any of the NASA centers the agency name is shown in conjunction with the full center
name (John F. Kennedy Space Center) as shown below.

<StoryContainer>
    <AgencyName size="md" centerName="John F. Kennedy Space Center" />
</StoryContainer>

### On light background

Against a light background, the agency name should be shown in black.

<StoryContainer>
    <AgencyName size="md" centerName="John F. Kennedy Space Center" />
</StoryContainer>

### On dark background

Against a very dark color background, the agency name should always be shown in white.

<StoryContainer background="dark">
    <AgencyName size="md" centerName="John F. Kennedy Space Center" theme="light" />
</StoryContainer>

### API

```javascript
import { AgencyName } from 'nasa-1975';
```

<Props of={AgencyName} />

Any other properties supplied will be spread to the root element.
