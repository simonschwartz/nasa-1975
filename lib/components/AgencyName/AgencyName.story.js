import React from "react";
import { text, select, boolean } from "@storybook/addon-knobs";
import { withInfo } from "@storybook/addon-info";
import AgencyName from "./";

const AgencyNameStory = withInfo({
  text:
    "The full agency name (National Aeronautics and Space Administration) can be displayed stacked or inline (defaults to stacked on small screens). To identify any of the NASA centers, the full center name is shown in conjunction with the full agency name."
})(() => {
  // Setup knob for setting stacked state
  const labelStacked = "stacked";
  const defaultValueStacked = false;
  const stackedValue = boolean(labelStacked, defaultValueStacked);

  // Setup knob for setting heading level
  const headingLevelLevel = "headingLevel";
  const optionsHeadingLevel = {
    1: "1",
    2: "2",
    3: "3",
    4: "4",
    5: "5"
  };
  const defaultValueHeadingLevel = "1";
  const headingLevelValue = select(
    headingLevelLevel,
    optionsHeadingLevel,
    defaultValueHeadingLevel
  );

  // Setup knob for setting heading size
  const labelSize = "size";
  const optionsSize = {
    lg: "lg",
    md: "md",
    sm: "sm"
  };
  const defaultValueSize = "lg";
  const sizeValue = select(labelSize, optionsSize, defaultValueSize);

  // Setup knob for setting center name text
  const labelCenterName = "centerName";
  const defaultValueCenterName = "";
  const centerNameValue = text(labelCenterName, defaultValueCenterName);

  return (
    <AgencyName
      stacked={stackedValue}
      headingLevel={headingLevelValue}
      size={sizeValue}
      centerName={centerNameValue}
    />
  );
});

export default AgencyNameStory;
