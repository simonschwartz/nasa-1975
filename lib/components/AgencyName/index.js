import AgencyName from "./AgencyName";

export default AgencyName;
