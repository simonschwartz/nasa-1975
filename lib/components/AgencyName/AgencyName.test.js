import React from "react";
import ReactDOM from "react-dom";
import AgencyName from "./";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<AgencyName />, div);
});
