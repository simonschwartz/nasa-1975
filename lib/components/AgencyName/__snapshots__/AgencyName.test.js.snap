// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct markup with centerName prop 1`] = `
<div
  className=""
>
  <h1
    className="NASA1975-heading NASA1975-heading--lg NASA1975-heading--light NASA1975-agency-name NASA1975-agency-name--inline"
  >
    <span
      className="NASA1975-agency-name-N"
    >
      National
    </span>
     
    <span
      className="NASA1975-agency-name-Aa"
    >
      Aeronautics and
    </span>
     
    <span
      className="NASA1975-agency-name-SA"
    >
      <span
        className="NASA1975-agency-name-S"
      >
        Space
      </span>
       
      <span
        className="NASA1975-agency-name-A"
      >
        Administration
      </span>
    </span>
  </h1>
  <h1
    className="NASA1975-heading NASA1975-heading--lg NASA1975-heading--medium "
  >
    John F. Kennedy Space Center
  </h1>
</div>
`;

exports[`renders correct markup with headingLevel prop 1`] = `
<div
  className=""
>
  <h3
    className="NASA1975-heading NASA1975-heading--lg NASA1975-heading--light NASA1975-agency-name NASA1975-agency-name--inline"
  >
    <span
      className="NASA1975-agency-name-N"
    >
      National
    </span>
     
    <span
      className="NASA1975-agency-name-Aa"
    >
      Aeronautics and
    </span>
     
    <span
      className="NASA1975-agency-name-SA"
    >
      <span
        className="NASA1975-agency-name-S"
      >
        Space
      </span>
       
      <span
        className="NASA1975-agency-name-A"
      >
        Administration
      </span>
    </span>
  </h3>
  
</div>
`;

exports[`renders correct markup with multiple props 1`] = `
<div
  className=""
>
  <h4
    className="NASA1975-heading NASA1975-heading--sm NASA1975-heading--light NASA1975-agency-name NASA1975-agency-name--stacked"
  >
    <span
      className="NASA1975-agency-name-N"
    >
      National
    </span>
     
    <span
      className="NASA1975-agency-name-Aa"
    >
      Aeronautics and
    </span>
     
    <span
      className="NASA1975-agency-name-SA"
    >
      <span
        className="NASA1975-agency-name-S"
      >
        Space
      </span>
       
      <span
        className="NASA1975-agency-name-A"
      >
        Administration
      </span>
    </span>
  </h4>
  <h4
    className="NASA1975-heading NASA1975-heading--sm NASA1975-heading--medium "
  >
    John F. Kennedy Space Center
  </h4>
</div>
`;

exports[`renders correct markup with no props 1`] = `
<div
  className=""
>
  <h1
    className="NASA1975-heading NASA1975-heading--lg NASA1975-heading--light NASA1975-agency-name NASA1975-agency-name--inline"
  >
    <span
      className="NASA1975-agency-name-N"
    >
      National
    </span>
     
    <span
      className="NASA1975-agency-name-Aa"
    >
      Aeronautics and
    </span>
     
    <span
      className="NASA1975-agency-name-SA"
    >
      <span
        className="NASA1975-agency-name-S"
      >
        Space
      </span>
       
      <span
        className="NASA1975-agency-name-A"
      >
        Administration
      </span>
    </span>
  </h1>
  
</div>
`;

exports[`renders correct markup with size prop 1`] = `
<div
  className=""
>
  <h1
    className="NASA1975-heading NASA1975-heading--md NASA1975-heading--light NASA1975-agency-name NASA1975-agency-name--inline"
  >
    <span
      className="NASA1975-agency-name-N"
    >
      National
    </span>
     
    <span
      className="NASA1975-agency-name-Aa"
    >
      Aeronautics and
    </span>
     
    <span
      className="NASA1975-agency-name-SA"
    >
      <span
        className="NASA1975-agency-name-S"
      >
        Space
      </span>
       
      <span
        className="NASA1975-agency-name-A"
      >
        Administration
      </span>
    </span>
  </h1>
  
</div>
`;

exports[`renders correct markup with stacked prop 1`] = `
<div
  className=""
>
  <h1
    className="NASA1975-heading NASA1975-heading--lg NASA1975-heading--light NASA1975-agency-name NASA1975-agency-name--stacked"
  >
    <span
      className="NASA1975-agency-name-N"
    >
      National
    </span>
     
    <span
      className="NASA1975-agency-name-Aa"
    >
      Aeronautics and
    </span>
     
    <span
      className="NASA1975-agency-name-SA"
    >
      <span
        className="NASA1975-agency-name-S"
      >
        Space
      </span>
       
      <span
        className="NASA1975-agency-name-A"
      >
        Administration
      </span>
    </span>
  </h1>
  
</div>
`;
